source/includes/examples/_cart-pole.html.md

```diff
@@ -91,37 +91,44 @@ This curriculum contains one lesson, called `balancing`. It configures the simul
 ## Simulator File
 
 ```python
-import gym
+import sys
+import logging
+from bonsai_ai import Brain, Config
+from bonsai_gym_common import GymSimulator
 
-import bonsai
-from bonsai_gym_common import GymSimulator, logging_basic_config
+log = logging.getLogger('gym_simulator')
+log.setLevel(logging.DEBUG)
 
-ENVIRONMENT = 'CartPole-v0'
-RECORD_PATH = None
 
+class CartPole(GymSimulator):
+    # Environment name, from openai-gym
+    environment_name = 'CartPole-v0'
 
-class CartPoleSimulator(GymSimulator):
+    # simulator name from Inkling
+    simulator_name = 'cartpole_simulator'
```

Comment:
Setting environment_name and simulator_name here is a bit unorthodox, and hardcodes this class to the exact inkling used in the example. In other words, this class will only work with inkling files that all happen to refer to the simulator by the exact same name. This will inhibit sharing of simulator code unless consumers also always either change their inkling to match the sim attribute, or change the sim attribute to match the inkling.

The more typical approach is to pass values as arguments to functions and constructors, as SDK1 did. Then it becomes trivial to dynamically specify the simulator name to match the inkling, if needed. Not that we expect huge amounts of dynamic changes, but this goes back to how simulators and drivers are actually two different concepts, like in SDK1.

Reply:
Yes, you mentioned this in the original code review. As I have pointed out in the past "CartPoleSimulator" is a class whose business is to simulate 'cartpole_simulator' inkling. It is a leaf example and I can't think of a compelling reason to pass an extra argument in when in practice it has one value: 'cartpole_simulator'